From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: schr-0dinger <kmkailasnath@gmail.com>
Date: Tue, 20 Oct 2026 05:23:54 +0700
Subject: [PATCH 3/9] imsmanager: Read the TTY type without SemCscFeature

getTtyType() asks Samsung's CSC (SemCscFeature, Samsung framework only)
whether the call app supports RTT and maps the profile's tty_type
between the TTY and RTT variants accordingly. Return the profile's
tty_type as is.

Assisted-by: LLM
---
 smali/com/sec/ims/settings/ImsProfile.smali | 88 ++-------------------
 1 file changed, 6 insertions(+), 82 deletions(-)

diff --git a/smali/com/sec/ims/settings/ImsProfile.smali b/smali/com/sec/ims/settings/ImsProfile.smali
index 028cb51..c5461c2 100644
--- a/smali/com/sec/ims/settings/ImsProfile.smali
+++ b/smali/com/sec/ims/settings/ImsProfile.smali
@@ -9472,95 +9472,19 @@
 .end method
 
 .method public getTtyType()I
-    .locals 4
+    .locals 1
 
-    .line 1486
-    invoke-static {}, Lcom/samsung/android/feature/SemCscFeature;->getInstance()Lcom/samsung/android/feature/SemCscFeature;
+    const-string v0, "tty_type"
 
-    move-result-object v0
+    invoke-virtual {p0, v0}, Lcom/sec/ims/settings/ImsProfile;->getAsInteger(Ljava/lang/String;)Ljava/lang/Integer;
 
-    const-string v1, "CscFeature_VoiceCall_SupportRTT"
+    move-result-object v0
 
-    invoke-virtual {v0, v1}, Lcom/samsung/android/feature/SemCscFeature;->getBoolean(Ljava/lang/String;)Z
+    invoke-virtual {v0}, Ljava/lang/Integer;->intValue()I
 
     move-result v0
 
-    .line 1487
-    .local v0, "isRttSupportByCallApp":Z
-    const-string v1, "tty_type"
-
-    invoke-virtual {p0, v1}, Lcom/sec/ims/settings/ImsProfile;->getAsInteger(Ljava/lang/String;)Ljava/lang/Integer;
-
-    move-result-object v1
-
-    invoke-virtual {v1}, Ljava/lang/Integer;->intValue()I
-
-    move-result v1
-
-    .line 1488
-    .local v1, "ttyType":I
-    new-instance v2, Ljava/lang/StringBuilder;
-
-    invoke-direct {v2}, Ljava/lang/StringBuilder;-><init>()V
-
-    const-string v3, "isRttSupportByCallApp : "
-
-    invoke-virtual {v2, v3}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
-
-    invoke-virtual {v2, v0}, Ljava/lang/StringBuilder;->append(Z)Ljava/lang/StringBuilder;
-
-    const-string v3, " ttyType : "
-
-    invoke-virtual {v2, v3}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
-
-    invoke-virtual {v2, v1}, Ljava/lang/StringBuilder;->append(I)Ljava/lang/StringBuilder;
-
-    invoke-virtual {v2}, Ljava/lang/StringBuilder;->toString()Ljava/lang/String;
-
-    move-result-object v2
-
-    const-string v3, "ImsProfile"
-
-    invoke-static {v3, v2}, Landroid/util/Log;->i(Ljava/lang/String;Ljava/lang/String;)I
-
-    .line 1489
-    if-eqz v0, :cond_1
-
-    const/4 v2, 0x1
-
-    if-eq v1, v2, :cond_0
-
-    const/4 v2, 0x2
-
-    if-ne v1, v2, :cond_1
-
-    .line 1490
-    :cond_0
-    add-int/lit8 v2, v1, 0x2
-
-    return v2
-
-    .line 1491
-    :cond_1
-    if-nez v0, :cond_3
-
-    const/4 v2, 0x3
-
-    if-eq v1, v2, :cond_2
-
-    const/4 v2, 0x4
-
-    if-ne v1, v2, :cond_3
-
-    .line 1492
-    :cond_2
-    add-int/lit8 v2, v1, -0x2
-
-    return v2
-
-    .line 1494
-    :cond_3
-    return v1
+    return v0
 .end method
 
 .method public getUse183OnProgressIncoming()Z
